### PR TITLE
introduction to sets

### PR DESCRIPTION
**Pull request by gigatesseract**
Added set data structure 

---

## ✅️ By submitting this PR, I have verified the following

- [x] Added descriptive name to PR
  - Your PR should NOT be called `Update index.md` since it does not help the maintainer understand what has been changed.
  - For example, if you create a **Variables** article inside the **Python** directory, the pull request title should be **Python: add Variables article**.
  - Other examples are **Git: edit Git Commit article** or **PHP: create PHP section and add Data Structures article**
- [x] Reviewed necessary formatting guidelines in [`CONTRIBUTING.md`](https://github.com/freeCodeCamp/guides/blob/master/CONTRIBUTING.md).
- [x] No plagiarized, duplicate or repetitive  content that has been directly copied from another source.

### CODE REVIEW

File: src/pages/cplusplus/set/index.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+More formally speaking, Sets are a type of associative containers in which each element has to be unique.
+* The value of the element cannot be modified once it is entered, although deleting an element and inserting a new element is allowed, the same way we do in mathenatics.
+* Set data sructure can be used to model, well, sets itself. It becomes easy to find intersections, unions etc.
+* Similar to vector, but only unique values are allowed.
+* Set arranges the elements in increasing order as and when you insert elements into the set. 
+
+The header file required for using the set data structure is 'set'. i.e, `#include<set>` must be there in your code for you to use the set data structure.    <br> 
+_Pro tip_:- Use `#include<bits/stdc++.h>` to include all C++ data structures and functions, instead of adding them one by one.    
+<br>
```
[the remark this answers is not in the document]
Comment (gigatesseract) [reply src/pages/cplusplus/set/index.md]: Okay. Thanks for letting me know. Ive resolved this.

File: src/pages/cplusplus/set/index.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ 
+  
+    return 0; 
+  
+}
+```
+
+Output:- 
+
+```
```
[the remark this answers is not in the document]
Comment (gigatesseract) [reply src/pages/cplusplus/set/index.md]: I have recitified it.

File: src/pages/cplusplus/set/index.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+The contents of myset : 	9	20	30	65	80
+
+myset after removal of elements less than 30 : 	30	65	80
+1 is removed
```
[the remark this answers is not in the document]
Comment (gigatesseract) [reply src/pages/cplusplus/set/index.md]: I understand. I have explained that functionality in the code itself. The formatting was pretty poor in my output.

File: src/pages/cplusplus/set/index.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
+---
+# title: Set
```
[the remark this answers is not in the document]
Comment (gigatesseract) [reply src/pages/cplusplus/set/index.md]: I have fixed it. Thank you.

**Comment by Asjas** [comment]
Thank you for your contribution to the page! :+1:
We're happy to accept these changes, and look forward to future contributions. :tada:

**Pull request merged**